src/commands.cc

```diff
@@ -1107,7 +1108,12 @@ KeymapMode parse_keymap_mode(const String& str)
     if (prefix_match("user", str))   return KeymapMode::User;
     if (prefix_match("object", str)) return KeymapMode::Object;
 
-    throw runtime_error(format("unknown keymap mode '{}'", str));
+    auto it = std::find(user_modes.begin(), user_modes.end(), str);
+    if (it == user_modes.end())
+        throw runtime_error(format("unknown keymap mode '{}'", str));
+
+    char offset = static_cast<char>(KeymapMode::Object) + 1;
```

Comment:
I think we should introduce a `KeymapMode::FirstUserMode` as the last KeymapMode, and use that rather than rely on the fact that `KeymapMode::Object` is the last one. With this code we'll get a bug as soon as a new non-user mode gets added at the end of the KeymapMode enum.